### ISSUE TITLE
Add support for edition 2023

### ISSUE THREAD

**Issue by friscoMad** [grpc/grpc-kotlin#625]
I am unsure how to do that in this plugin, in grpc-java it was done in [this commit](https://github.com/grpc/grpc-java/commit/5322414505c61dfcdf538f38dfe467df9dab3739)

This is the error with 1.4.1:
XXX: is an editions file, but code generator protoc-gen-grpckt hasn't been updated to support editions yet.  Please ask the owner of this code generator to add support or switch back to proto2/proto3

**Comment by ejona86** [comment]
grpc-java doesn't have support yet either. While the code can compile to support additions, the protobuf version actually used for the binaries on Maven Central is before editions was enabled. Since I don't see any C++ files for the generator in grpc-kotlin and I know it makes use of protoc-gen-grpc-java, I suspect your issue is actually the same as https://github.com/grpc/grpc-java/issues/11526 (yes, I see you created this one first).

**Comment by ejona86** [comment]
Actually, it looks like the plugin needs to [include the features supported in its response to protoc](https://github.com/protocolbuffers/protobuf/blob/9e1fdb0c015aeb584cf36780cd0564f26adf932e/src/google/protobuf/compiler/plugin.cc#L140). The current code has some plumbing for that, but doesn't mention editions:
https://github.com/grpc/grpc-kotlin/blob/a969a91ca37867fa28d83fca234a2b4ced7b1540/compiler/src/main/java/io/grpc/kotlin/generator/protoc/CodeGenerators.kt#L77-L80

So I think you are right that grpc-kotlin needs an update here. But you're _also_ blocked on the grpc-java bug.

**Comment by friscoMad** [comment]
@ejona86 You are right I was pretty sure that I was able to build a edition 2023 version of our services with the latest grpc-java but I have tried again and as reported it fails.
Still, adding support in grpc-kotlin should be easy, and it is a requirement anyway, so there is no reason for not making the change.

**Comment by vladimirfx** [comment]
Java support Editions 2024 now. This becomes blocker for migration for Kotlin based projects.

**Comment by lockhartja** [comment]
Hey smart people -- any chances of picking this up?

**Comment by friscoMad** [comment]
After seeing no movement on this for years, I did create my own grpc-kotlin generator that generates code that does not depend on grpc-java (no more java + kotlin sources), supports editions up to 2024 and it is a binary plugin instead of a jar one so it is a lot faster (for small proto sources) and easier to use outside the gradle/maven plugins.
Plus some extra goodies like optional comments included in generated code or deprecate annotations for deprecated methods/services.

Generated code still uses this project runtime libraries and maintains the same API, so moving to it should be as simple as replacing grpc-java and grpc-kotlin plugins with kotpb-grpc-codegen

https://github.com/Kotpb/kotpb-grpc-codegen
